components/sync_manager/src/manager.rs

```diff
@@ -30,42 +30,71 @@ impl SyncManager {
         SyncEngineId::try_from(engine_name).map_err(SyncManagerError::UnknownEngine)
     }
 
-    fn get_engine(engine_id: &SyncEngineId) -> Option<Box<dyn SyncEngine>> {
+    // Interrupt any active sync operations
+    pub fn interrupt(&self) {
+        InterruptScope::interrupt();
+        places::interrupt_current_sync_query();
```

Comment:
eg, I can see here how the places sync connection is interrupted, but what do we do for other engines? How does their sql connection get interrupted?